core/store/src/flat/store_helper.rs

```diff
@@ -174,3 +174,73 @@ pub fn key_belongs_to_shard(
     let (key_shard_uid, _) = decode_flat_state_db_key(key)?;
     Ok(key_shard_uid.version == shard_layout.version() && key_shard_uid.shard_id as u64 == shard_id)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::iter_flat_state_entries;
+    use crate::flat::store_helper::{set_delta, set_flat_state_value};
+    use crate::flat::{
+        BlockInfo, FlatStateChanges, FlatStateDelta, FlatStateDeltaMetadata, FlatStateValue,
+        FlatStorageManager,
+    };
+    use crate::test_utils::create_test_store;
+    use crate::CryptoHash;
+    use near_primitives::shard_layout::ShardUId;
+
+    #[test]
+    fn test_iter_flat_state_entries() {
+        // Setup shards and store
+        let store = create_test_store();
+        let shard_uids = Vec::from([
+            ShardUId { shard_id: 0, version: 0 },
+            ShardUId { shard_id: 1, version: 0 },
+            ShardUId { shard_id: 2, version: 0 },
+        ]);
+        let root = CryptoHash::default();
+        let flat_storage_manager = FlatStorageManager::test(store.clone(), &shard_uids, root);
+
+        for (i, shard_uid) in shard_uids.iter().enumerate() {
+            let mut store_update = store.store_update();
+            let key: Vec<u8> = vec![0, 1, i as u8];
+            let val: Vec<u8> = vec![0, 1, 2, i as u8];
+
+            // Add value to flat store
+            assert!(set_flat_state_value(
+                &mut store_update,
+                *shard_uid,
+                key.clone(),
+                Some(FlatStateValue::Inlined(val.clone())),
+            )
+            .is_ok());
+
+            let new_root = CryptoHash::default();
```

Comment:
`root` can be confusing here since we also have a concept of state root which is also a hash. I think replacing root with `block` would make it less confusing